File: app/src/main/java/com/bq/zowi/models/commands/DataRequestCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class DataRequestCommand extends MotionlessCommand {
    private static final String APP_ID = "I";
    private static final String BATTERY = "B";
    private static final Command.Action[] DATA_REQUEST_ACTIONS = {Command.Action.DISTANCE, Command.Action.NOISE, Command.Action.BATTERY, Command.Action.APP_ID, Command.Action.NAME};
    private static final String DISTANCE = "D";
    private static final String NAME = "E";
    private static final String NOISE = "N";

    public DataRequestCommand(Command.Action action) {
        setAction(action);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return DATA_REQUEST_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public String getCommandValue() {
        String commandValue;
        switch (getAction()) {
            case DISTANCE:
                commandValue = DISTANCE;
                break;
            case NOISE:
                commandValue = NOISE;
                break;
            case BATTERY:
                commandValue = BATTERY;
                break;
            case APP_ID:
                commandValue = APP_ID;
                break;
            case NAME:
                commandValue = NAME;
                break;
            default:
                throw new IllegalStateException("Action not matching any valid value");
        }
        return commandValue + Command.CRLN;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new DataRequestCommand(getAction());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/DataRequestCommand.kt
package com.bq.zowi.models.commands

class DataRequestCommand(action: Command.Action) : MotionlessCommand() {

    init {
        setAction(action)
    }

    override fun getAllowedActions(): Array<Command.Action> = DATA_REQUEST_ACTIONS
    override fun getCommandValue(): String = when (getAction()) {
        Command.Action.DISTANCE -> "D"
        Command.Action.NOISE -> "N"
        Command.Action.BATTERY -> "B"
        Command.Action.APP_ID -> "I"
        Command.Action.NAME -> "E"
        else -> throw IllegalStateException("Action not matching any valid value")
    } + Command.CRLN

    override fun copy(): Command = DataRequestCommand(getAction())

    companion object {
        private val DATA_REQUEST_ACTIONS = arrayOf(
            Command.Action.DISTANCE, Command.Action.NOISE, Command.Action.BATTERY,
            Command.Action.APP_ID, Command.Action.NAME
        )
    }
}
